content/cucumber/browser-automation.md

```diff
@@ -16,86 +17,149 @@ Automation tools such as:
 
 ## Selenium WebDriver
 
-Let's convert the [Selenium-Webdriver by Example tutorial](http://docs.seleniumhq.org/docs/03_webdriver.jsp#introducing-the-selenium-webdriver-api-by-example)
-to use Cucumber.
+WebDriver is designed to provide a simpler, more concise programming interface in addition to addressing some limitations in the Selenium-RC API. Selenium-WebDriver was developed to better support dynamic web pages where elements of a page may change without the page itself being reloaded. WebDriver's goal is to supply a well-designed object-oriented API that provides improved support for modern advanced web-app testing problems.
 
-We can express the example as the following Scenario:
+Let us look at an example of Cucumber using selenium-webdriver in UI testing, by converting [Selenium-Web driver by example     ](http://docs.seleniumhq.org/docs/03_webdriver.jsp#introducing-the-selenium-webdriver-api-by-example).
 
-```gherkin
-Scenario: Finding some cheese
-  Given I am on the Google search page
-  When I search for "Cheese!"
-  Then the page title should start with "cheese"
-```
+ We can express the example as the following Scenario:
 
-Here are the accompanying Step Definitions:
-
-[carousel]
-
-```ruby
-# TODO! See the Java example for now.
+```Gherkin
+Scenario: Finding some cheese
+   Given I am on the Google search page
+   When I search for "Cheese!"
+   Then the page title should start with "cheese"
 ```
 
 ```java
-package com.example;
+package class.exmple;
 
 public class ExampleSteps {
-    private final WebDriver driver = new FirefoxDriver();
 
-    @Given("^I am on the Google search page$")
+    private final WebDriver driver = new FirefoxDriver();
+    @Given("^I am on the Google search page$"\)
     public void I_visit_google() {
-        driver.get("https://www.google.com");
-    }
+    driver.get("https:\\www.google.com");
+   }
 
-    @When("^I search for \"(.*)\"$")
-    public void search_for(String query) {
-        WebElement element = browser.findElement(By.name("q"));
-        // Enter something to search for
+   @When("^I search for \"(.*)\"$")
+     public void search_for(String query) {
+        WebElement element = browser.findElelment(By.name("q"));
+        \\Enter Something to search for
```

Comment:
A single line comment in Java should be preceded by '//' not '\\'
(Note: replace multiple times!)